pytorch_lightning/trainer/training_loop.py

```diff
@@ -524,7 +524,7 @@ def run_training_epoch(self):
             self.increment_accumulated_grad_global_step()
 
             # max steps reached, end training
-            if self.max_steps is not None and self.max_steps == self.global_step:
+            if self.max_steps is not None and self.max_steps >= self.global_step:
```

Comment:
```suggestion
            if self.max_steps is not None and self.max_steps <= self.global_step:
```
pretty sure this is what you mean, since global step is the variable that increases and max_steps is the one that is constant. 
Besides that, for an easier understanding, i would just swap the variables around

Reply:
nonetheless, I think this fix in my suggestion is correct even if it does not fix what you orignially had in mind.